docs/architecture.md

```diff
@@ -23,67 +23,87 @@
 
 ## Core components
 
-### Connections
+### Protocol
 
-A connection allows the AEA to connect to an external service which has a Python SDK or API. A connection wraps an external SDK or API.
+Protocols define how messages are represented and encoded for transport. They also define the rules to which messages have to adhere in a message sequence. 
 
-### Skill
+For instance, a protocol may contain messages of type `START` and `FINISH`. From there, the rules could prescribe that a message of type `FINISH` must be preceded by a message of type `START`.
 
-A skill can encapsulate any code and ideally delivers economic value to the AEA. Each skill has at most a single Handler and potentially multiple Behaviours and Tasks. The Handler is responsible for dealing with messages of the protocol type for which this skill is registered, as such it encapsulates `reactions`. A Behaviour encapsulates `actions`, that is sequences of interactions with other agents initiated by the AEA. Finally, a Task encapsulates background work which is internal to the AEA.
+The `Message` class in the `protocols/base.py` module provides an abstract class with all the functionality a derived Protocol message class requires for a custom skill, such as basic message generating and management functions and serialisation details.
 
-### Protocol
+### Connection
 
-Protocols define how messages are represented and encoded for transport. They also define the rules to which messages have to adhere in a message sequence. For instance, a protocol might have a message of type START and FINISH. Then the rules could prescribe that a message of type FINISH must be preceded by a message of type START.
+A connection wraps an external SDK or API and manages the messaging. It allows the agent to connect to an external service which has a Python SDK or API. 
+
+### Skill
 
+Skills deliver economic value to the AEA by allowing an agent to encapsulate and call any kind of code. It encapsulates Handlers, Behaviours, and Tasks.
 
-### Mailbox
+* Handler: each skill has a single Handler which is responsible for the registered protocol messaging. By understanding the requirements contained in Envelopes, the Handler reacts appropriately to message requests.
+* Behaviour: one or more Behaviours encapsulate sequences of actions that cause interactions with other agents initiated by the framework. 
+* Task: one or more Tasks encapsulate background work internal to the agent.
 
-TBC.
+### MailBox
+
+A MailBox contains InBox and OutBox queues which manage Envelopes.
 
 ### Envelope
```

Comment:
I think this should be the first section as these are the 'events' around which the whole framework centers